charts/templates/statefulset.yaml

```diff
@@ -21,6 +21,45 @@ spec:
       affinity: {{- toYaml . | nindent 8}}
       {{- end }}
       terminationGracePeriodSeconds: {{ .Values.terminationGracePeriod }}
+      {{- if ne .Values.bootstrapHostName "" }}
+      initContainers: 
+      - name: check-group-init-container
+        image: "{{ .Values.image.repository }}:{{ .Values.image.tag }}"
+        command:
+          - bash
+          - '-c'
+          - |
+            echo '### Begin init container ###'
+
+            GROUP_CFG_TEMPLATE='{"group-name":"%s"}'
+            GROUP_CFG=$(printf "$GROUP_CFG_TEMPLATE" "$MARKLOGIC_GROUP")
+
+            GROUP_RESP_CODE=`curl --anyauth -m 20 -s -o /dev/null -w "%{http_code}" -X GET http://${BOOTSTRAP_HOST_NAME}:8002/manage/v2/groups/${MARKLOGIC_GROUP} --anyauth --user ${MARKLOGIC_ADMIN_USERNAME}:${MARKLOGIC_ADMIN_PASSWORD}`
+            
+            if [[ ${GROUP_RESP_CODE} -eq 200 ]]; then
+              echo "Group $MARKLOGIC_GROUP is already configured on the MarkLogic cluster." 
+            else
+              echo "Group $MARKLOGIC_GROUP does not exist, creating $MARKLOGIC_GROUP on the MarkLogic cluster"
+              curl --anyauth --user ${MARKLOGIC_ADMIN_USERNAME}:${MARKLOGIC_ADMIN_PASSWORD} -m 20 -s -X POST -d "${GROUP_CFG}" -H "Content-type: application/json" http://${BOOTSTRAP_HOST_NAME}:8002/manage/v2/groups
+              sleep 20s
```

Comment:
20s might be too long for create a new group. As init-container is blocking the main container, we might want to reduce it to 5s.